src/map_extras.cpp

```diff
@@ -1206,98 +1213,99 @@ static bool mx_minefield( map &m, const tripoint &abs_sub )
         //Set two warning signs on the last horizontal line of the submap
         x = rng( 3, SEEX );
         x1 = rng( SEEX + 1, SEEX * 2 - 4 );
-        m.furn_set( point( x, SEEY * 2 - 1 ), furn_str_id( "f_sign_warning" ) );
-        m.set_signage( tripoint( x, SEEY * 2 - 1, abs_sub.z ), text );
-        m.furn_set( point( x1, SEEY * 2 - 1 ), furn_str_id( "f_sign_warning" ) );
-        m.set_signage( tripoint( x1, SEEY * 2 - 1, abs_sub.z ), text );
+        tm.furn_set( point( x, SEEY * 2 - 1 ), furn_str_id( "f_sign_warning" ) );
+        tm.set_signage( tripoint( x, SEEY * 2 - 1, abs_sub.z ), text );
+        tm.furn_set( point( x1, SEEY * 2 - 1 ), furn_str_id( "f_sign_warning" ) );
+        tm.set_signage( tripoint( x1, SEEY * 2 - 1, abs_sub.z ), text );
 
         did_something = true;
     }
 
-    if( bridge_at_south && bridge_at_center && road_at_north ) {
+    if( bridge_at_south && bridgehead_at_center && road_at_north ) {
+        tm.load( omt_to_sm_copy( abs_omt + point_north ), false );
```

Comment:
And by the way: you can change the parameter name of the map to something else (`mx_minefield( map &m_, ...)` and than you can use the name `m` for your local `tinymap`. That allows the remaining code to be kept unchanged, it can still refer to `m`, only that `m` is now the `tinymap`.

That would make the diff view much smaller.

Reply:
Fixed, thanks.